packages/fxa-settings/src/components/ConnectedServices/Service.tsx

```diff
@@ -0,0 +1,132 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import React from 'react';
+import { LinkExternal } from 'fxa-react/components/LinkExternal';
+import { DeviceLocation } from '../../models/Account';
+import { ReactComponent as WebIcon } from './web.svg';
+import { ReactComponent as DesktopIcon } from './desktop.svg';
+import { ReactComponent as TabletIcon } from './tablet.svg';
+import { ReactComponent as MobileIcon } from './mobile.svg';
+import { ReactComponent as LockwiseIcon } from './lockwise.svg';
+import { ReactComponent as PocketIcon } from './pocket.svg';
+import { ReactComponent as MonitorIcon } from './monitor.svg';
+import { ReactComponent as SyncIcon } from './sync.svg';
+import { ReactComponent as FPNIcon } from './fpn.svg';
+
+export function Service({
+  name,
+  deviceType,
+  location,
+  lastAccessTimeFormatted,
+  canSignOut,
+}: {
+  name: string;
+  deviceType: string;
+  location: DeviceLocation;
+  lastAccessTimeFormatted: string;
+  canSignOut: boolean;
+}) {
+  const { city, stateCode, country } = location;
+  const locationProvided = Boolean(city && stateCode && country);
+  const isPocket = name === 'Pocket';
+  const isMonitor = name === 'Firefox Monitor';
+  const isLockwise = name === 'Firefox Lockwise';
+  const isSync = name === 'Firefox Sync';
+  const isFPN = name === 'Firefox Private Network';
+
+  const renderLink = isPocket || isMonitor || isLockwise || isSync || isFPN;
+  let serviceLink = '';
+  let iconSvg = <WebIcon data-testid="web-icon" />;
+
+  if (renderLink) {
+    if (isPocket) {
+      serviceLink = 'https://www.mozilla.org/en-US/firefox/pocket/';
+      iconSvg = <PocketIcon data-testid="pocket-icon" />;
+    }
+    if (isMonitor) {
+      serviceLink = 'https://monitor.firefox.com/';
+      iconSvg = <MonitorIcon data-testid="monitor-icon" />;
+    }
+    if (isLockwise) {
+      serviceLink = 'https://www.mozilla.org/en-US/firefox/lockwise/';
+      iconSvg = <LockwiseIcon data-testid="lockwise-icon" />;
+    }
+    if (isSync) {
+      serviceLink =
+        'https://support.mozilla.org/en-US/kb/how-do-i-set-sync-my-computer';
+      iconSvg = <SyncIcon data-testid="sync-icon" />;
+    }
+    if (isFPN) {
+      serviceLink = 'https://vpn.mozilla.com/';
+      iconSvg = <FPNIcon data-testid="fpn-icon" />;
```

Comment:
What happens if none of the `if` clauses are truthy? could that happen?

Lines 32-63 are one of those rare situations where I think it'd be cleaner to have a bunch of little FooService subclasses that implement a basic Service interface, rather than sprinkling conditional logic all over the place. That's a purely optional refactoring though.

Reply:
If none are true then we get the default web icon. Also in this case the "service" isn't actually a service and just a device, so we don't provide a link